share/tools/create_manpage_completions.py

```diff
@@ -866,15 +866,25 @@ def get_paths_from_manpath():
         parent_paths = os.getenv("MANPATH").strip().split(':')
     else:
         # Some systems have manpath, others have `man --path` (like Haiku).
-        # TODO: Deal with systems that have neither (OpenBSD)
-        for prog in [['manpath'], ['man', '--path']]:
-            try:
-                proc = subprocess.Popen(prog, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-            except OSError: # Command does not exist, keep trying
-                continue
-            break # Command exists, use it.
-        manpath, err_data = proc.communicate()
-        parent_paths = manpath.decode().strip().split(':')
+        if os.uname()[0] == 'OpenBSD':
+            parent_paths = ['/usr/share/man', '/usr/X11R6/man', '/usr/local/man']
+            if os.path.isfile('/etc/man.conf'):
```

Comment:
Does anyone else use another incompatible /etc/man.conf?

If nobody does, we can just try this after we've tried to call the tools.

Reply:
I am not entirely sure if anyone else uses it.

Reply:
I can't find anything on centos (which would be Red Hat), but it seems that FreeBSD and macOS use it as well.

See [the FreeBSD manpage](https://www.unix.com/man-page/freebsd/5/man.conf/). Given that that uses a directive called "MANPATH" where [OpenBSD uses "manpath"](http://man.openbsd.org/man.conf.5), we could just try reading either (i.e. "the word manpath, case-insensitively") after we've exhausted the programs (which should still be preferred because they're more likely to get weird edgecases right). If that still doesn't work, we could still fallback to the current default.

As you might have guessed by now, I have a certain philosophy when it comes to cross-platform support, in that I really don't like checking for the specific platform name via `uname` or similar. I very much prefer feature-testing if at all possible. This allows us to e.g. do proper completion for GNU utilities even on systems where they aren't usually installed. (One interesting case where we can't do that is `killall`, because we absolutely cannot invoke it on Solaris - because it actually kills all)